timely/examples/columnar.rs

```diff
@@ -331,6 +332,11 @@ mod builder {
             self.empty = self.pending.pop_front();
             self.empty.as_mut()
         }
+
+        #[inline]
+        fn relax(&mut self) {
+            *self = Self::default();
+        }
```

Comment:
Should we maybe `debug_assert` that this is empty? That's the intent, right? It should be a semantic no-op, but potentially release resources.